ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Wiki/WikisSample.cs

```diff
@@ -0,0 +1,217 @@
+            VssConnection connection = this.Context.Connection;
+            WikiHttpClient wikiClient = connection.GetClient<WikiHttpClient>();
+
+            List<WikiV2> wikis = wikiClient.GetAllWikisAsync().SyncResult();
+
+            foreach (WikiV2 wiki in wikis)
+            {
+                Console.WriteLine("Retrieved wiki with name '{0}' in project '{1}'", wiki.Name, wiki.ProjectId);
+            }
+
+            return wikis;
+        }
+
+        [ClientSampleMethod]
+        public WikiV2 UpdateWiki()
+        {
+            VssConnection connection = this.Context.Connection;
+            GitHttpClient gitClient = connection.GetClient<GitHttpClient>();
+            WikiHttpClient wikiClient = connection.GetClient<WikiHttpClient>();
+            
+            // Get all the existing wikis
+            List<WikiV2> wikis = wikiClient.GetAllWikisAsync().SyncResult();
+
+            // Get the code wiki for which we need to update the versions
+            WikiV2 codeWiki = wikis.Where(wiki => wiki.Type == WikiType.CodeWiki).FirstOrDefault();
+            
+            if (codeWiki == null)
+            {
+                Console.WriteLine("No code wiki to continue the update operation.");
+            
+                return null;
+            }
+
+            // Get the versions in that wiki
+            List<GitVersionDescriptor> versions = codeWiki.Versions.ToList();
+
+            // Append the new version
+            List<GitBranchStats> branches = gitClient.GetBranchesAsync(codeWiki.ProjectId, codeWiki.RepositoryId).SyncResult();
+            foreach(var branch in branches)
+            {
+                versions.Add(new GitVersionDescriptor()
+                {
+                    Version = branch.Name
+                });
+            }
+
+            WikiUpdateParameters updateParams = new WikiUpdateParameters()
+            {
+                Versions = versions
+            };
+
+            WikiV2 updatedCodeWiki = wikiClient.UpdateWikiAsync(updateParams, codeWiki.ProjectId, codeWiki.Name).SyncResult();
+
+            Console.WriteLine("Updated wiki with name '{0}' in project '{1}'", updatedCodeWiki.Name, updatedCodeWiki.ProjectId);
+            Console.WriteLine("Updated versions are : {0}", string.Join(",", updatedCodeWiki.Versions.Select(v => v.Version)));
+            
+            return updatedCodeWiki;
+        }
+
+        [ClientSampleMethod]
+        public WikiV2 DeleteCodeWiki()
+        {
+            VssConnection connection = this.Context.Connection;
+            GitHttpClient gitClient = connection.GetClient<GitHttpClient>();
+            WikiHttpClient wikiClient = connection.GetClient<WikiHttpClient>();
+
+            // Get all the existing wikis
+            List<WikiV2> wikis = wikiClient.GetAllWikisAsync().SyncResult();
+
+            // Get the code wiki for which we need to update the versions
+            WikiV2 codeWiki = wikis.Where(wiki => wiki.Type == WikiType.CodeWiki).FirstOrDefault();
```

Comment:
Instead of randomly deleting the first wiki you find (which can be really bad if a user runs this sample against their work or personal accounts), have this method create a new code wiki and then delete it. This ensures your sample isn't destructive (to a legitimate resource). This is also a more legitimate sample since there's almost never a case where it makes sense to iterate over a list of wikis, randomly pick one, and delete it :)